### ISSUE TITLE
x/tools/gopls: unused completions after "." ".." or "........"

### ISSUE THREAD

**Issue by netjune** [golang/go#37358]
example code:

```
package "main"

main () {
    .|
}
```

### What version of Go are you using (`go version`)?
1.13.5

### Does this issue reproduce with the latest release?
Yes

### What operating system and processor architecture are you using (`go env`)?
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/Users/jun/Library/Caches/go-build"
GOENV="/Users/jun/Library/Application Support/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GONOPROXY=""
GONOSUMDB=""
GOOS="darwin"
GOPATH="/Users/jun/Projects/go"
GOPRIVATE=""
GOPROXY="https://goproxy.io,direct"
GOROOT="/usr/local/Cellar/go/1.13.5/libexec"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/Cellar/go/1.13.5/libexec/pkg/tool/darwin_amd64"
GCCGO="gccgo"
AR="ar"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
GOMOD="/Users/jun/Projects/go/src/golang.org/x/tools/gopls/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/yl/0xyrgw396fj89vd1n2vk649c0000gn/T/go-build115939707=/tmp/go-build -gno-record-gcc-switches -fno-common"

**Comment by muirdm** [comment]
I can guess what the issue is about, but it would still be helpful to answer the issue template questions.

"." alone is a syntax error so gopls falls back to lexical completions. This case is more noticeable than others since "." triggers completions in most editors. However, for example, triggering completions after a lone "(" results in the same thing:

```go
package "main"
main () {
    (
}
```

**Comment by netjune** [comment]
> I can guess what the issue is about, but it would still be helpful to answer the issue template questions.

Sorry for my poor English. It happens mainly in the following case:
```
pkt := append(pkt, buf[:cnt]...|)
```
Cursor is at '|'.

> "." alone is a syntax error so gopls falls back to lexical completions. This case is more noticeable than others since "." triggers completions in most editors. However, for example, triggering completions after a lone "(" results in the same thing:
> 
> ```go
> package "main"
> main () {
>     (
> }
> ```

This doesn't popups completion menu here(emacs+lsp-mode).

**Comment by netjune** [comment]
Another related issue:
```
package "main"
main () {
}

aaaaaa|
```
Cursor is at "|". There are many completion items. And they don't have prefix "aaaaaa".

**Comment by muirdm** [comment]
> This doesn't popups completion menu here(emacs+lsp-mode).

Run `(company-manual-begin)` to trigger a completion. "." is a character that triggers completions automatically.

> Cursor is at "|". There are many completion items. And they don't have prefix "aaaaaa".

That is a separate issue and is actually fixed on master. Completions at the file level scope are special (only keywords are allowed).

**Comment by netjune** [comment]
> > This doesn't popups completion menu here(emacs+lsp-mode).
> 
> Run `(company-manual-begin)` to trigger a completion. "." is a character that triggers completions automatically.
> 

Yes.

> > Cursor is at "|". There are many completion items. And they don't have prefix "aaaaaa".
> 
> That is a separate issue and is actually fixed on master. Completions at the file level scope are special (only keywords are allowed).

Thanks.

**Comment by muirdm** [comment]
> It happens mainly in the following case:
> `pkt := append(pkt, buf[:cnt]...|)`

My first reaction is that this is an editor bug. Editors should not trigger completion after multiple ".", only after a single one. That being said, VSCode seems to also pop up completions automatically after "...", so perhaps it is worth suppressing completions when the cursor is directly after the "...". @stamblerre what do you think?

**Comment by zikaeroh** [comment]
I don't think it's unusual for the editor to do completions like that, given you declare that it should send completions so after a `.` (with no other qualifier).

If I were typing `.` or `..`, a completion of `...` would be handy in specific contexts, and there are probably other languages out there which would want similar things. An immediate example that's not exactly a `.` would be in C++ where you may declare to the client the intent to complete after `:` in order to get completions for namespaces after `::`. Not triggering for repeats would prevent that sort of completion.

I'd personally suppress after `...`.
